src/main/java/org/galatea/starter/domain/rpsy/IexHistoricalPricesKey.java

```diff
@@ -1,36 +1,52 @@
 package org.galatea.starter.domain.rpsy;
 
-import lombok.AccessLevel;
-import lombok.Builder;
-import lombok.NoArgsConstructor;
-import org.apache.commons.lang3.builder.HashCodeBuilder;
+import lombok.*;
+import org.galatea.starter.domain.RangeType;
+import org.galatea.starter.utils.Helpers;
 
 import java.io.Serializable;
+import java.time.LocalDate;
+import java.util.ArrayList;
 import java.util.Date;
+import java.util.List;
+import java.util.Optional;
 
+@Data
 @Builder
+@AllArgsConstructor(access = AccessLevel.PUBLIC)
 @NoArgsConstructor(access = AccessLevel.PUBLIC)
 public class IexHistoricalPricesKey implements Serializable {
     public String symbol;
     public Date date;
 
-    public IexHistoricalPricesKey(String symbol, Date date){
-        this.symbol = symbol;
-        this.date = date;
-    }
+    /**
+     * Returns a list of IexHistoricalPricesKeys for the given symbol and range.
+     * For example, a range of 5d will return keys for that symbol's last 5 days
+     *
+     * @param symbol stock symbol to get historical prices for.
+     * @param range the range of the historical data.
+     * @return a list of IexHistoricalPricesKeys
+     */
+    public static List<IexHistoricalPricesKey> keysForRange(final String symbol, final String range){
+        LocalDate sinceDate;
+        Integer rangeValue = Integer.parseInt(range.replaceAll("[^0-9]",""));
+
+        Optional<RangeType> type = RangeType.fromString(String.valueOf(range.charAt(range.length()-1)));
 
-    @Override
-    public boolean equals(Object obj) {
-        if(!(obj.getClass().equals(this.getClass()))){
-            return false;
+        if(type.isPresent()){
+            sinceDate = type.get().subtract(rangeValue);
+        }else{
+            throw new IllegalArgumentException();
```

Comment:
You should put an explanation of what caused the exception into the Exception itself -- an empty Exception wouldn't be helpful if you were to be woken up at 5am to a call from support :) 

`throw new IllegalArgumentException(String.format("Invalid range type specified: %s", type))`

You should put the value from the String.valueOf(range....) call into a variable so that you can log it out as above.

Reply:
Good point!